### PR TITLE
Fix generic method constraints

### LINKED ISSUE #8: Doesn't generate generic constraints in methods
Current version now works with namespaces. Tried it with my code and here we go again)) 

Class:

using AutomaticInterfaceAttribute;

namespace TestNuget;

[GenerateAutomaticInterface]
public class GenericTest : IGenericTest
{
    `public void Test<T>() where T : class { }`
}

Result:

using System.CodeDom.Compiler;
using AutomaticInterfaceAttribute;

namespace TestNuget
{
    [GeneratedCode("AutomaticInterface", "")]
    public partial interface IGenericTest
    {
        void Test();
    }
}

---

GenericTest.cs(6, 28): [CS0535] 'GenericTest' does not implement interface member 'IGenericTest.Test()'

Will try to find fix in couple days

### PR DESCRIPTION
**Pull request by G1Vh**
fix #8

### CODE REVIEW

File: AutomaticInterface/AutomaticInterface/RoslynExtensions.cs [hunk 1]
```diff
@@ -71,6 +68,118 @@ public static List<string> GetUsings(this CompilationUnitSyntax root)
...
+            string result = "";
+
+            if (typeArgs.Any())
+            {
+                result += "<";
+
+                bool isFirstIteration = true;
+                foreach (ITypeSymbol typeArg in typeArgs)
+                {
+                    // insert comma if not first iteration
```
Comment (ChristianSauer) [inline AutomaticInterface/AutomaticInterface/RoslynExtensions.cs:148]: maybe first create a list of strToAdd and then do a String.Join? Could be more readable

File: AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs [hunk 2]
```diff
@@ -112,7 +113,24 @@ public string Build()
             foreach (var method in methodInfos)
             {
                 cb.AppendAndNormalizeMultipleLines(method.Documentation);
```
Comment (ChristianSauer) [inline AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs:115]: I would probably put this foreach body into a new method. The method gets too long

**Review by ChristianSauer — CHANGES_REQUESTED** [review]
I like this very much, I totally forgot to check for where constraints. Makes me wonder what else I missed.

I would also like an added example in Readme.md - Example.

Note that I am on vacation from tomorrow until the 14th, so creating a new release in that timeframe is impossible

File: AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs (hunk 2 above)
Comment (G1Vh) [reply AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs]: Extracted it

File: AutomaticInterface/AutomaticInterface/RoslynExtensions.cs (hunk 1 above)
Comment (G1Vh) [reply AutomaticInterface/AutomaticInterface/RoslynExtensions.cs]: Rewrited

**Comment by G1Vh** [comment]
Vacation is great! 
I will use local package if encounter any bugs and fix it in that time, don't worry.

**Review by ChristianSauer — COMMENTED** [review]
Looks good, I am merging

**Pull request merged**